### ISSUE TITLE
MacOS Monterey 12.0.1 : Zim window not appearing on launch

### ISSUE THREAD

**Issue by rasisin** [zim-desktop-wiki/zim-desktop-wiki#1712]
**Describe the bug**
When launching Zim in the current version of MacOS (Monterey 12.0.1), Python launches but there is no Zim window anywhere. I have tried adjusting the display resolution and that does not solve the problem.
Zim had been working fine on the same machine previously.
Terminal shows the error message 'WARNING: XDG_RUNTIME_DIR is not set...'
I don't know if the problem was caused by the OS upgrade, or by some dependency being updated.

**To Reproduce**
1. Type 'zim' + ENTER from terminal command line.
2. notice an error message in the terminal: in my case, 'WARNING: XDG_RUNTIME_DIR is not set, using /var/folders/8_/jggm7f2950s9zf6qbjv0m_l00000gn/T/zim-user1 as a fallback'
3. notice that Python launches and the Zim menu appears at the top of the screen, but the actual Zim window never appears. 
4. Select menu items that launch a new window, like 'View Debug Log' or 'Preferences'. Notice that the top bar of the window appears, but not the window itself.

**Environment (please complete the following information):**
 - Version: 0.74.2
 - OS: MacOS
 - Language: English (US)

**Comment by dehesselle** [comment]
The `XDG_RUNTIME_DIR` warning is only a warning and won't be the cause of the issue.
Can you test if https://gitlab.com/dehesselle/zim_macos/-/releases has the same issue? (I'm not running Monterey yet.)

**Comment by rasisin** [comment]
That one works fine, thank you!

It has a couple of minor issues in case you have time:
1. It doesn't seem to be using the style.conf file from the Homebrew install of Zim. Does it have its own style.conf file somewhere? Or some other way of adjusting that?
2. Most of the plugins work, but Equation Editor is not recognising the dependencies like Latex that I had installed via Homebrew for Zim there. Any thoughts on how to fix this?

**Comment by dehesselle** [comment]
1. This app is independent from Homebrew (by design), that's why it's not using it.
I'm not familiar with `style.conf`, but I presume you're looking for `$HOME/Library/Application Support/org.zim-wiki.Zim/zim/style.conf`?
2. There is a way to configure that. First test the following: open Terminal, make sure the Latex binaries are in your `$PATH` and then start Zim from the Terminal by running `/Applications/Zim.app/Contents/MacOS/Zim`. Does the equation editor work now?
   - Yes: You can persist this configuration by creating a file `$HOME/Library/Application Support/org.zim-wiki.Zim/zimapp.ini` and add the following there:
     ```ini
     [environment]
     PATH=<add the complete output from 'echo $PATH' here>:/Applications/Zim.app/Contents/Resources/bin
     ```
     The equation editor should work now when you launch Zim the regular way.
   - No: I'll need to investigate. Where did you install Homebrew to (default is `/usr/local` on x86, `/opt/homebrew` on arm)? Latex install command (blind guess: `brew install latex`)? Anything else or special I should know about?

**Comment by rasisin** [comment]
1. --> Thank you!
2. --> Yes --> it works, thank you!

**Comment by dehesselle** [comment]
Awesome, thank you for testing @rasisin

**Comment by murattepegoz** [comment]
@dehesselle this is amazing, I was not aware of your macos installer. Once it is official, it will be very useful for mac users.
One quick question: style.conf folder you shared in the previous message is working, however, I could not find where to store gtk-3.0 settings.ini. Do you have any idea? 
`$HOME/Library/Application Support/org.zim-wiki.Zim/gtk-3.0` folder does exist, but settings.ini inside this folder seems to be not working.

**Comment by dehesselle** [comment]
`$HOME/Library/Application Support/org.zim-wiki.Zim/gtk-3.0/settings.ini` is correct and will be used.

You're probably trying to set the theme (`gtk-theme-name`), which is the one setting that is currently not respected/internally overwritten. This will be fixed in the next release. In the meantime, use something like this in `$HOME/Library/Application Support/org.zim-wiki.Zim/zimapp.ini`:
```ini
[environment]
GTK_THEME=Adwaita
```

**Comment by kjyv** [comment]
So, while I'm also using the installer in the mean time (thanks!), back to the OP issue:
Starting zim -D, I get the following output, maybe it's helpful:

```
INFO: Can not import 'xdg.Mime' - falling back to 'mimetypes'
INFO: Can not import 'xdg.Mime' - falling back to 'mimetypes'
DEBUG: Loading config from: <ConfigFile: /Users/stefan/.config/zim/preferences.conf>
DEBUG: New extendable: <zim.plugins.InsertedObjectTypeMap object at 0x106cd7580>
WARNING: XDG_RUNTIME_DIR is not set, using /var/folders/70/62mngyv158b4rrgbvz41y3g80000gn/T/zim-stefan as a fallback
DEBUG: Loading plugin: pageindex
DEBUG: Loading plugin: insertsymbol
DEBUG: Loading plugin: printtobrowser
DEBUG: Loading plugin: osx_menubar
INFO: This is zim 0.74.2
DEBUG: Python version is sys.version_info(major=3, minor=9, micro=7, releaselevel='final', serial=0)
DEBUG: Platform is posix
DEBUG: Not running from a source dir
DEBUG: Set XDG_DATA_HOME to /Users/stefan/.local/share
DEBUG: Set XDG_DATA_DIRS to [<Dir: /usr/local/share>, <Dir: /usr/local/Cellar/zim/0.74.2/libexec/share>]
DEBUG: Set XDG_CONFIG_HOME to /Users/stefan/.config
DEBUG: Set XDG_CONFIG_DIRS to [<Dir: /etc/xdg>]
DEBUG: Set XDG_CACHE_HOME to /Users/stefan/.cache
DEBUG: Connecting to /var/folders/70/62mngyv158b4rrgbvz41y3g80000gn/T/zim-stefan/zim-0.74.2-679b97e9
DEBUG: Got error in dispatch: No such file or directory
DEBUG: Starting primary process
DEBUG: Start listening on: /var/folders/70/62mngyv158b4rrgbvz41y3g80000gn/T/zim-stefan/zim-0.74.2-679b97e9
DEBUG: Loading config from: <zim.notebook.info.VirtualFile object at 0x10aca8730>
DEBUG: Loading config from: /Users/stefan/(...)/notes/Zim/notebook.zim
DEBUG: Loading config from: /Users/stefan/(...)/notes/Zim/notebook.zim
DEBUG: Connecting to database file: /Users/stefan/.cache/zim/(...)/index.db
DEBUG: Loading config from: /Users/stefan/.cache/zim/(...)/state.conf
DEBUG: New extendable: <Notebook: Notes>
DEBUG: Loading config from: <ConfigFile: /Users/stefan/.config/zim/style.conf>
DEBUG: Autosave interval: 10 - use threads: True
INFO: Page changed on disk: To Do
INFO: Open page: To Do (To Do)
DEBUG: New extendable: <notebookview.NotebookView object at 0x10acf7880 (zim+gui+notebookview+NotebookView at 0x7fe0a688dad0)>
DEBUG: Load extension: <class 'zim.plugins.insertsymbol.InsertSymbolPageViewExtension'>
DEBUG: Load extension: <class 'zim.plugins.pageindex.PageIndexNotebookViewExtension'>
DEBUG: Action: toggle_panes(False)
DEBUG: Action: toggle_panes(True)
DEBUG: Load extension: <class 'zim.plugins.printtobrowser.PrintToBrowserPageViewExtension'>
DEBUG: New extendable: <mainwindow.MainWindow object at 0x10accd200 (zim+gui+mainwindow+MainWindow at 0x7fe0a7084260)>
DEBUG: Load extension: <class 'zim.plugins.osx_menubar.OSXMenuBarMainWindowExtension'>

(zim:94166): Gtk-WARNING **: 15:25:23.095: toggle_toolbar: missing action toggle_toolbar

(zim:94166): Gtk-WARNING **: 15:25:23.096: toggle_statusbar: missing action toggle_statusbar

(zim:94166): Gtk-WARNING **: 15:25:23.096: toolbar_menu: missing action toolbar_menu

(zim:94166): Gtk-WARNING **: 15:25:23.099: reload_index: missing action reload_index

(zim:94166): Gtk-WARNING **: 15:25:23.105: set_toolbar_style_icons_and_text: missing action set_toolbar_style_icons_and_text

(zim:94166): Gtk-WARNING **: 15:25:23.105: set_toolbar_style_icons_only: missing action set_toolbar_style_icons_only

(zim:94166): Gtk-WARNING **: 15:25:23.105: set_toolbar_style_text_only: missing action set_toolbar_style_text_only

(zim:94166): Gtk-WARNING **: 15:25:23.105: set_toolbar_icon_size_large: missing action set_toolbar_icon_size_large

(zim:94166): Gtk-WARNING **: 15:25:23.105: set_toolbar_icon_size_small: missing action set_toolbar_icon_size_small

(zim:94166): Gtk-WARNING **: 15:25:23.105: set_toolbar_icon_size_tiny: missing action set_toolbar_icon_size_tiny
WARNING: /usr/local/Cellar/zim/0.74.2/libexec/lib/python3.9/site-packages/zim/plugins/osx_menubar.py:111: Warning: invalid cast from 'GtkMenuBar' to 'GtkWindow'
  _global_osx_application.set_menu_bar(self.menubar)


(zim:94166): Gtk-CRITICAL **: 15:25:23.107: gtk_window_add_accel_group: assertion 'GTK_IS_WINDOW (window)' failed
DEBUG: Accelmap: /Users/stefan/.config/zim/accelmap
DEBUG: Add window: MainWindow
DEBUG: BackgroundCheck started
DEBUG: BackgroundCheck finished
DEBUG: BackgroundCheck started
DEBUG: BackgroundCheck finished
```

**Comment by reagle** [comment]
As discussed on the list, I've encountered something similar:

> I just upgraded my office M1 Mac Mini to Monterey and all I can see is the title bar (attached). I made no other change.
> 
> `arch --x86_64 /usr/local/bin/python3.9 ~/bin/zim-desktop-wiki/zim.py ~/joseph/plan/zim/ --standalone --geometry 900x1000-0-0`
> 
> After noticing this, I pulled the latest from github (commit 41905b7ef026d10f8df2b9f2143bf8412990d98c). The problem persists.
> 
> Falling back to 0.7.35 gives me a black box under the title bar instead of a transparent region (attached).

**Comment by dehesselle** [comment]
I believe you guys are all hitting the same issue and we should label this "upstream".
GTK up to the current 3.24.30 is broken on Monterey. It has already been fixed, but 3.24.31 has not been released yet. If my Zim.app works for you too, @reagle , that'll be further confirmation as my version comes with a patched GTK 3.24.30.

**Comment by reagle** [comment]
@dehesselle thanks for taking on Mac packaging; that'll be fantastic to have. I'll next be on my Monterey machine on Friday and will try it then. (BTW: It's working great on the Big Sur machine I'm on now.)

**Comment by reagle** [comment]
@dehesselle where is your Apps config settings? I don't think it's reading `~/.config/zim` is it?

**Comment by dehesselle** [comment]
@reagle I'm following Apple standards, you'll find that in `~/Library/Application Support/org.zim-wiki.Zim`.

**Comment by dehesselle** [comment]
I'm closing this since the official Zim.app (check the website / download section if you haven't 😉) works on Monterey.

I'm sorry if this causes you inconvenience, but I cannot offer support for Homebrew or MacPorts.

**Comment by reagle** [comment]
@dehesselle, do you have a preference of where discussion of homebrew should happen? (Here, Gitlab, at the Homebrew project)? I know you won't support Homebrew, but for those of us who use it, we should discuss what should be done with the [existing brew](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/zim.rb). I use Homebrew, but don't know if it should be deleted or set as x86 only? Also,
as long as your are providing a versioned dmg or zip file, [creating and maintaining a cask](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md) is straightforward and is something others could do.

**Comment by dehesselle** [comment]
From the looks of it, I think we should use https://github.com/jaap-karssenberg/zim-wiki/wiki . There are already various things documented in regards to macOS, we could restructure that a bit so that there's a macOS section with various sub pages where we then consolidate all the tips and instructions.

**Comment by reagle** [comment]
That's fine for documentation, but where do the homebrew folk discuss strategy?

**Comment by dehesselle** [comment]
I guess you have to educate me on how you've been doing this before. Are you thinking about using issues or discussions or ...? Also, when you say "homebrew folk", is it more about random people showing up or do we have regulars here?

I'm not opposed to talking about Homebrew or MacPorts (I simply cannot take on more work), I want to avoid collecting issues if we know nobody is ever going to work on them and that it's just sheer luck if they ever get any response - that's frustrating for both users and developers.

From the top of my head, I'd say "talking strategy" is good for discussions, with the mid- and long-term "results" being documented in the wiki. But please, tell me how you would want to do this.

**Comment by rasisin** [comment]
Question from an unsophisticated user:
I find the .DMG installer satisfactory. Is there some advantage to using Homebrew for Zim on Mac?

**Comment by reagle** [comment]
@rasisin, there's no need to use Homebrew if you don't want to. I like it because I get package updates automatically.

@dehesselle, I'm no homebrew expert, just a user. And I know casks are relatively straightforward if a versioned zip or dmg is already provided. I've submitted requests to homebrew when packages have been updated. I'm going to open a new issue in this project for those who might be interested.
